packages/data-service/src/data-service.ts

```diff
@@ -300,37 +332,120 @@ class DataService extends EventEmitter {
+   * Compared to the listCollections, this method will never throw
+   *
+   * TODO: Only needed while we are waiting for authorizedCollections: true
+   * support to land in the nodejs driver
+   */
+  listCollectionsNamesOnly(
+    dbName: string,
+    callback: Callback<ReturnType<typeof adaptCollectionInfo>[]>
+  ): void {
+    const listCollectionsAsync = promisify(this.listCollections.bind(this)) as (
+      // Promisify doesn't extract correct types from the method because of the
+      // overloads so we have to convince it that the arguments we provide are
+      // fine
+      dbName: string,
+      filter: Document,
+      options: { nameOnly: true; authorizedCollections?: true }
+    ) => Promise<ReturnType<typeof adaptCollectionInfo>[]>;
+
+    void Promise.all([
+      getDatabasesAndCollectionsFromPrivileges(this._initializedClient, [
+        'find',
+      ]).catch(() => ({ collections: [] })),
+      listCollectionsAsync(
+        dbName,
+        {},
+        {
+          nameOnly: true,
+          // TODO: This does nothing right now, see note above
+          authorizedCollections: true,
+        }
+      ).catch(() => []),
+    ]).then(([{ collections }, collectionsInfo]) => {
+      if (collectionsInfo.length > 0) {
+        callback(null, collectionsInfo);
+        return;
+      }
+      if (collections.length > 0) {
+        callback(
+          null,
+          collections.map((name) => adaptCollectionInfo({ name, db: dbName }))
+        );
+        return;
+      }
+      callback(null, []);
+    });
+  }
+
   /**
    * List all databases on the currently connected instance.
    *
    * @param callback - The callback.
    */
-  listDatabases(callback: Callback<Document>): void {
+  listDatabases(
+    callback: Callback<ReturnType<typeof adaptDatabaseInfo>[]>
+  ): void;
+
+  listDatabases(
+    options: { nameOnly: true },
+    callback: Callback<ReturnType<typeof adaptDatabaseInfo>[]>
+  ): void;
+
+  listDatabases(
+    optionsOrCallback:
+      | { nameOnly: true }
+      | Callback<ReturnType<typeof adaptDatabaseInfo>[]>,
+    callback?: Callback<ReturnType<typeof adaptDatabaseInfo>[]>
+  ): void {
+    let options: { nameOnly: true } | null = null;
+
+    if (typeof optionsOrCallback === 'function') {
+      callback = optionsOrCallback;
+    } else {
+      options = optionsOrCallback;
+    }
+
     const logop = this._startLogOp(
       mongoLogId(1_001_000_033),
-      'Running listDatabases'
+      'Running listDatabases',
+      { nameOnly: options?.nameOnly ?? false }
     );
-    this._initializedClient.db('admin').command(
-      {
-        listDatabases: 1,
-      },
-      {
-        readPreference: this.getReadPreference(),
-      },
-      (error, result) => {
-        logop(error);
-        if (error) {
-          // @ts-expect-error Callback without result...
-          return callback(this._translateMessage(error));
+
+    this._initializedClient
```

Comment:
as above .. if we are changing the only place where this call is done maybe we can just move to async / await so we won't have to re-change this code later, but feel free to keep it

Reply:
Not much! Just me lacking some context, thanks for the suggestion!

Reply:
Done in d67891d75cfb28248fd4e821971d89c1dbe07902